ferrilata-devilline/Services/Extensions/AdminJsonAnalizer.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ferrilata_devilline.Services.Extensions
+{
+    public static class AdminJsonAnalizer
+    {
+        private class ExpectedFormat
+        {
+            [JsonProperty(Required = Required.Always)]
+            string version { set; get; }
+
+            [JsonProperty(Required = Required.Always)]
+            string name { set; get; }
+
+            [JsonProperty(Required = Required.Always)]
+            string tag { set; get; }
+
+            [JsonProperty(Required = Required.Always)]
+            List<object> levels { set; get; }
+
+            public ExpectedFormat()
+            {
+            }
+        }
+
+        public static bool HasMissingFieldsAsAdmin(this JObject data)
+        {
+            try
```

Comment:
Using exceptions for control flow is generally [considered](https://docs.microsoft.com/en-us/visualstudio/profiling/da0007-avoid-using-exceptions-for-control-flow?view=vs-2019) bad practice.

Reply:
ok! Will try to simplify